x-pack/plugins/security_solution/server/lib/detection_engine/rules/utils.ts

```diff
@@ -325,6 +325,7 @@ export const legacyMigrate = async ({
   const [siemNotification, legacyRuleActionsSO] = await Promise.all([
     rulesClient.find({
       options: {
+        filter: 'alert.attributes.alertTypeId:(siem.notifications)',
```

Comment:
NOTE: Without this filter of `filter: 'alert.attributes.alertTypeId:(siem.notifications)',`, we could have been getting other rules that happen to have the mentioned `rule.id` as part of their reference. At the moment I think `siem.notifications` is the only rule that happens to mention another rule within its SO references, but later if we do add rules that have references to other rules then this would become a problem.

Using this filter, we narrow down the type which should avoid the issue or questions around it.